server.js

```diff
@@ -34,29 +36,41 @@ const userMessageTimes = {};
 
 /* Handle incoming messages */
 bot.on("message", msg => {
-    const query = msg.content.substr(commandChar.length).split(" ");
-    const command = query[0].toLowerCase();
-    const parameter = query.length > 1 ? query.slice(1).join(" ") : "";
+    // generate RegExp pattern for message parsing
+    // Example: ^!(card|price) ?(.*)$|!(card|price) ?([^!]*)(!|$)
+    const charPattern = _.escapeRegExp(commandChar);
+    const commandPattern = charPattern+'('+Object.keys(handlers).map(_.escapeRegExp).join('|')+')';
+    const regExpPattern = `^${commandPattern} ?(.*)$|${commandPattern} ?([^${charPattern}]*)(${charPattern}|$)`;
```

Comment:
Ooh right, that's a remnant from when it was just the beginning of the line and me overthinking things again. Fixed.